Dockerfile

```diff
@@ -0,0 +1,21 @@
+# Build the binary
+FROM golang:1.14.1 as builder
+
+WORKDIR /workspace
+# Copy the Go Modules manifests
+COPY go.mod go.mod
+COPY go.sum go.sum
+# cache deps before building and copying source so that we don't need to re-download as much
+# and so that source changes don't invalidate our downloaded layer
+RUN go mod download
+
+# Copy the go source
+COPY . .
+
+# Build
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -o applicationset-controller main.go
+
+# Use distroless as minimal base image to package the manager binary
+FROM debian:10-slim
```

Comment:
can we use alpine?

Reply:
of course.

I copy this from argo-cd Dockerfile.

I think it's ok to use alpine.

Reply:
Better yet would be `scratch`.

Also, we should run as a non-root, numeric user (i.e. 999).
```Dockerfile
USER 999
```

See https://github.com/argoproj/argo-cd/pull/3108 for rationale.